lib/matplotlib/legend.py

```diff
@@ -1141,27 +1149,19 @@ def _get_legend_handles(axs, legend_handler_map=None):
     for ax in axs:
         handles_original += [
             *(a for a in ax._children
-              if isinstance(a, (Line2D, Patch, Collection))),
+              if isinstance(a, (Line2D, Patch, Collection, Text))),
             *ax.containers]
         # support parasite axes:
         if hasattr(ax, 'parasites'):
             for axx in ax.parasites:
                 handles_original += [
                     *(a for a in axx._children
-                      if isinstance(a, (Line2D, Patch, Collection))),
+                      if isinstance(a, (Line2D, Patch, Collection, Text))),
                     *axx.containers]
 
-    handler_map = Legend.get_default_handler_map()
-
-    if legend_handler_map is not None:
-        handler_map = handler_map.copy()
-        handler_map.update(legend_handler_map)
-
-    has_handler = Legend.get_legend_handler
-
     for handle in handles_original:
         label = handle.get_label()
-        if label != '_nolegend_' and has_handler(handler_map, handle):
+        if label not in ['_nolegend_', '']:
```

Comment:
We currently do the filtering to "non empty labels" in https://github.com/matplotlib/matplotlib/blob/3e32cc7e88d6d583942335dff4c83b4b7dd13ae9/lib/matplotlib/legend.py#L1139-L1152 it is probably better if we leave that filtering step there and continue to only do "artist noped out of being in the legend" here.